website/blog/2024-06-26-llm-as-judge/index.md

```diff
@@ -0,0 +1,522 @@
+        completion = openai.chat.completions.create(
+            model="gpt-3.5-turbo",
+            messages=[
+                {"role": "system", "content": system_prompt},
+                {"role": "user", "content": f"{row}"},
+            ],
+        )
+        answers.append(completion.choices[0].message.content)
+
+    return answers
+```
+
+Then our evaluation is similar to what we did before
+
+```python
+with mlflow.start_run() as run:
+    results = mlflow.evaluate(
+        my_llm,
+        eval_data,
+        model_type="text",
+        evaluators="default",
+        extra_metrics=[faithfulness],
+        evaluator_config={
+        "col_mapping": {
+            "inputs": "llm_inputs",
+           }}
+    )
+mlflow.end_run()
+```
+
+### GenAI Metrics
+
+As an alternative we can use mlflow built in metrics for genai using the same examples.
+
+MLflow offers a few pre-built metrics which uses LLM as the judge. Despite the difference under the hood, the usage is the same - put these metrics in the extra_metrics argument in mlflow.evaluate(). Here is the list of pre-built metrics:
+
+1. **[mlflow.metrics.genai.answer_similarity()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.answer_similarity)**
+
+2. **[mlflow.metrics.genai.answer_correctness()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.answer_correctness)**
+
+
+3. **[mlflow.metrics.genai.answer_relevance()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.answer_relevance)**
+ 
+4. **[mlflow.metrics.genai.relevance()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.relevance)**
+
+5. **[mlflow.metrics.genai.faithfulness()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.faithfulness)**
+
+Let's use mlflow's pre-built in faithfulness metric.
+
+
+```python
+from mlflow.metrics.genai import EvaluationExample, faithfulness
+faithfullness_metric = faithfulness(model="openai:/gpt-4")
+print(faithfullness_metric)
+```
+
+This metric works quite well in singery with langchain retrivals since you can provide the grading context seperatly from the llm_input column if you prefer.
+
+Since in this example we are doing everything in the same input column let's map out the context column to our input column.
+
+
+
+```python
+with mlflow.start_run() as run:
+    results = mlflow.evaluate(
+        my_llm,
+        eval_data,
+        model_type="text",
+        evaluators="default",
+        extra_metrics=[faithfullness_metric],
+        evaluator_config={
+        "col_mapping": {
+            "inputs": "llm_inputs",
+            "context": "llm_inputs",
+           }}
+    )
+mlflow.end_run()
+```
+
+![Gauge Faithfullness Chart](faithfulness.png)
+
+
+## Conclusion
+
```

Comment:
Include a note in the conclusion about how we might use evaluations to improve the model

Reply:
Added this paragraph on that, let me know if its similar to what you had in mind